[Bug] Crashing when tapping Add note

###### Reproduction Steps

1. Tap `Add Note`


###### Expected Result

The add note screen should appear


###### Actual Result

It crashes

###### Debug info
```
AnkiDroid Version = 2.16alpha48

Android Version = 11

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-N770F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.58 Mobile Safari/537.36

ACRA UUID = e7a03b0c-3285-489b-8bcd-aa1e128e2eb5

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [X] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] I have searched for similar existing issues here and on the user forum
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### Error log on android studio
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki, PID: 23259
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.NoteEditor}: com.ichi2.utils.JSONException: org.json.JSONException: Value null at did of type org.json.JSONObject$1 cannot be converted to long
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3846)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4022)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2336)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8653)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
     Caused by: com.ichi2.utils.JSONException: org.json.JSONException: Value null at did of type org.json.JSONObject$1 cannot be converted to long
        at com.ichi2.utils.JSONObject.getLong(JSONObject.java:345)
        at com.ichi2.anki.NoteEditor.setDid(NoteEditor.java:1834)
        at com.ichi2.anki.NoteEditor.onCollectionLoaded(NoteEditor.java:578)
        at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:384)
        at com.ichi2.anki.NoteEditor.onCreate(NoteEditor.java:438)
        at android.app.Activity.performCreate(Activity.java:8207)
        at android.app.Activity.performCreate(Activity.java:8191)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3819)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4022) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2336) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:246) 
        at android.app.ActivityThread.main(ActivityThread.java:8653) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130) 
     Caused by: org.json.JSONException: Value null at did of type org.json.JSONObject$1 cannot be converted to long
        at org.json.JSON.typeMismatch(JSON.java:101)
        at org.json.JSONObject.getLong(JSONObject.java:528)
        at com.ichi2.utils.JSONObject.getLong(JSONObject.java:343)
        at com.ichi2.anki.NoteEditor.setDid(NoteEditor.java:1834) 
        at com.ichi2.anki.NoteEditor.onCollectionLoaded(NoteEditor.java:578) 
        at com.ichi2.anki.AnkiActivity.startLoadingCollection(AnkiActivity.java:384) 
        at com.ichi2.anki.NoteEditor.onCreate(NoteEditor.java:438) 
        at android.app.Activity.performCreate(Activity.java:8207) 
        at android.app.Activity.performCreate(Activity.java:8191) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3819) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4022) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2336) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:246) 
        at android.app.ActivityThread.main(ActivityThread.java:8653) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130) 
D/UsageAnalytics: sendAnalyticsException() description/fatal: org.json.JSONException: Value null at did of type org.json.JSONObject$1 cannot be converted to long/true
I/ACRA: ACRA is disabled for com.ichi2.anki - forwarding uncaught Exception on to default ExceptionHandler
I/Process: Sending signal. PID: 23259 SIG: 9
Disconnected from the target VM, address: 'localhost:2478', transport: 'socket'
```

For some reason, the did on my collection was null, so it crashes at `getLong()`. I'll push the fix

![2022_03_05-14_41_59](https://user-images.githubusercontent.com/69634269/156894533-6fc78651-15e4-445b-84ad-855867c69a6f.png)

--- a/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java
@@ -1819,7 +1819,7 @@ public class NoteEditor extends AnkiActivity implements
             return;
         }
         if (note == null || mAddNote || mCurrentEditedCard == null) {
-            JSONObject model = getCol().getModels().current();
+            Model model = getCol().getModels().current();
             if (getCol().get_config("addToCur", true)) {
                 mCurrentDid = getCol().get_config_long(CURRENT_DECK);
                 if (getCol().getDecks().isDyn(mCurrentDid)) {
@@ -1831,7 +1831,7 @@ public class NoteEditor extends AnkiActivity implements
                     mCurrentDid = 1;
                 }
             } else {
-                mCurrentDid = model.getLong("did");
+                mCurrentDid = model.getDid();
             }
         } else {
             mCurrentDid = mCurrentEditedCard.getDid();

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java
@@ -870,7 +870,7 @@ public class Collection implements CollectionGetter {
                     due = (long) nextID("pos");
                 }
                 if (did == null || did == 0L) {
-                    did = model.getLong("did");
+                    did = model.getDid();
                 }
                 // add any missing cards
                 ArrayList<JSONObject> tmpls = _tmplsFromOrds(model, avail);

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/Model.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/Model.java
@@ -80,6 +80,11 @@ public class Model extends JSONObject {
         return getJSONArray("flds").getJSONObject(pos);
     }
 
+    /**
+     * @return model did or default deck id (1) if null
+     */
+    public Long getDid() { return isNull("did") ? 1L : getLong("did"); }
+
     public List<String> getTemplatesNames() {
         return getJSONArray("tmpls").toStringList("name");
     }

--- a/AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.java
@@ -673,4 +673,24 @@ public class ModelTest extends RobolectricTest {
         assertListEquals(Arrays.asList(0, 1), Models.availOrds(reverse, new String[]{"Foo", "Bar"}, Models.AllowEmpty.TRUE));
         assertListEquals(Arrays.asList(1), Models.availOrds(reverse, new String[]{"  \t ", "Bar"}, Models.AllowEmpty.TRUE));
     }
+
+    /**
+     * tests if Model.getDid() returns model did
+     * or default deck id (1) if null
+     */
+    @Test
+    public void getDid_test() {
+        Collection col = getCol();
+        ModelManager mm = col.getModels();
+        Model basic = mm.byName("Basic");
+        basic.put("did", 999L);
+
+        Long expected = 999L;
+        assertEquals("getDid() should return the model did", expected, basic.getDid());
+
+        // Check if returns default deck id (1) when did is null
+        basic.put("did", null);
+        Long expected2 = 1L;
+        assertEquals("getDid() should return 1 (default deck id) if model did is null", expected2, basic.getDid());
+    }
 }

